### PR TITLE
rpmb: move rpmb_mutex to fs layer

### PR DESCRIPTION
**Pull request by jforissier**
In rpmb_fs_write(), if other session write RPMB between read_fat() and
tee_rpmb_write(), it may write to the same address. So move rpmb_mutex
to fs layer for concurrency scenarios.

Signed-off-by: Lijianhui (Airbak) <airbak.li@hisilicon.com>
[Removed tee_rpmb_read_unlocked(), unlock earlier on a few locations]
Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

### CODE REVIEW

**Comment by jforissier** [comment]
Test case: https://github.com/OP-TEE/optee_test/pull/119

**Comment by jforissier** [comment]
@Airbak FYI

**Comment by airbak** [comment]
@jforissier thanks for you help :)

**Comment by jforissier** [comment]
Ping! Can anyone review this please?

**Comment by jenswi-linaro** [comment]
`rpmb_fs_close()` should use `rpmb_mutex` too.

File: core/tee/tee_rpmb_fs.c [hunk 1]
```diff
@@ -2261,6 +2247,8 @@ static tee_fs_off_t rpmb_fs_lseek(TEE_Result *errno, int fd,
 	if (!fh)
 		return TEE_ERROR_BAD_PARAMETERS;
 
+	mutex_lock(&rpmb_mutex);
```
Comment (jenswi-linaro) [inline core/tee/tee_rpmb_fs.c]: This should be done before `handle_lookup()` above.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jforissier** [comment]
@jenswi-linaro thanks

**Pull request merged**